### ISSUE TITLE
ActionService: Source of ActionFailure Event object unclear and what happens after sending the failure event

### ISSUE THREAD

**Issue by NicolasVorwerg** [SamCooper/MC_SPEC_RIDS#189]
The requirement 3.2.5.d is defined as: "The source of the ActionFailure event shall be the ActivityTracking Execution event that represents the final stage of the action execution." Please specify what is the "final state of the action execution. Is it the the latest successful progress-stage-event, the last progress-stage-event or is it the action-completion-event? 

Maybe the question is getting obsolete after answering another uncertainty, which is:
What are the events that will be published after there is a failure event and whats the order of these? I suppose any further execution is aborted and the failure event is the last event that will be received by the consumer. But there is no requirement saying so.

So in my understanding which makes sense most is:
- if there is no failure it would look for instance like this: 
- starting-event -> progress-stage-1-event ->  progress-stage-2-event -> progress-stage-3-event -> completion event.
- if there is a failure: 
- starting-event -> progress-stage-1-event ->  progress-stage-2-event -> {failure} -> failure-event (with source-link to progress-stage-2-event-object) (no more progress events and no completion event).

Is this the right behaviour? 
In summary please explicitly define which events will be send after the failure event (or that there are no more events send) and which event exactly is the failure-events source-link.

**Comment by SamCooper** [comment]
It is actually:
-  starting-event -> progress-stage-1-event -> progress-stage-2-event -> {failure} -> progress-stage-3-event(success:false) AND failure-event (with source-link to progress-stage-3-event-object) (no more progress events and no completion event).

I have updated the requirements to reflect this.
